### DISCUSSION TITLE
Super-user

### DISCUSSION THREAD

**Discussion by Rokulus** [gaganpreet/fastapi-starter#514] [General]
Hello, I started using your fastapi-starter and I would like to know where is login info for superuser or how to create superuser ?

When I try to authorize in API docs I always get error code 405 and I'am not sure how to authorize for using API. Or is API only accessible from UI ? 

I'am pretty new to react-admin and if you can redirect me to some spesific documentation or please clarify for me how to create superuser I would be very thankful

**Comment by gaganpreet** [comment]
> When I try to authorize in API docs I always get error code 405 and I'am not sure how to authorize for using API. Or is API only accessible from UI ?

Did you run the ["Step 1" scripts](https://github.com/gaganpreet/fastapi-starter/tree/main/%7B%7Bcookiecutter.project_slug%7D%7D#step-1-getting-started) after setting up the project?

To create a super user, you can follow these steps:

- Go to the frontend url in your browser (localhost:3000 by default)
- Sign up as a new user
- Connect to psql  `docker-compose exec postgres psql -U postgres app`
- Run this sql: `UPDATE users SET is_superuser = t WHERE email = '<your_email>'`

**Comment by Rokulus** [comment]
Thank you for your answer. 

But I still don't get why I can't authorize myself in docs for API. In authorize window I just write username as email address I log in with and password. 

Step I took:

1. I runned cookiecutter command `cookiecutter https://github.com/gaganpreet/fastapi-starter` and CD to created directory
2. I runned `docker-compose up -d` and everythin build fine
3. I runned `docker-compose exec backend alembic upgrade head` which gave back:
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 7e09fa75df7a, Add initial migration
INFO  [alembic.runtime.migration] Running upgrade 7e09fa75df7a -> 5c89a726934c, Add item migration
4. And I runned `docker-compose exec postgres createdb apptest -U postgres`

Then I go to frontend and I sign up. Then I go to API docs and when I want to authorize myself it will give back error code 405 (method not allowed). Even if I make the user to super_user and to verified it still gives me back 405. I'am not sure if I missed something

I'am just curious if this is desired behavior and if it is I will change it to my desires. Thank your for your answer in advance.

**Reply by gaganpreet** [reply]
> Then I go to API docs and when I want to authorize myself it will give back error code 405 (method not allowed).

Do you mean the `/api/v1/auth/jwt/login` endpoint? I tried it with a username and password following the same steps as you described, and I got back a JSON object an `access_token`.

> Even if I make the user to super_user and to verified it still gives me back 405. I'am not sure if I missed something

This doesn't work because only a super user can make another user a super user. But since you just signed up as the first user, no user is a super user y et. The first user you will only be able to make a super user with a raw SQL query.
 
> I'am just curious if this is desired behavior and if it is I will change it to my desires. Thank your for your answer in advance.

You're welcome! Hope this helps you understand what's going on. :slightly_smiling_face:

**Reply by wang-arthur** [reply]
I think I had the same issue as @Rokulus: not with the `/api/v1/auth/jwt/login` endpoint, which worked for me, but with the `/docs/auth/jwt/login` endpoint:

- go to the docs page / http://localhost:8000/docs/
- click the `Authorize` button (upper right)
- In the modal, enter username & password and click `Authorize Button`
- Get a 405 `Auth Error Error: Method Not Allowed`
